trace-to-events/src/main.rs

```diff
@@ -144,18 +150,53 @@ async fn main() -> anyhow::Result<()> {
                     process_kafka_message(
                         &tracer,
                         &args,
-                        &mut kafka_producer_thread_set,
+                        &sender,
                         &producer,
                         &m,
-                    );
+                    )?;
                     consumer.commit_message(&m, CommitMode::Async).unwrap();
                 }
                 Err(e) => warn!("Kafka error: {}", e)
-            },
-            join_next = kafka_producer_thread_set.join_next() => {
-                if let Some(Err(e)) = join_next {
-                    error!("Error Joining Kafka Producer Task: {e}");
+            }
+        }
+    }
+}
+
+type DigitiserEventListToBufferSender = Sender<DeliveryFuture>;
+type TrySendDigitiserEventListError = TrySendError<DeliveryFuture>;
+
+// The following functions control the kafka producer thread
+fn create_producer_task(
+    send_digitiser_eventlist_buffer_size: usize,
+) -> DigitiserEventListToBufferSender {
+    let (channel_send, channel_recv) =
+        tokio::sync::mpsc::channel::<DeliveryFuture>(send_digitiser_eventlist_buffer_size);
+
+    tokio::spawn(produce_to_kafka(channel_recv));
+    channel_send
+}
+
+async fn produce_to_kafka(mut channel_recv: Receiver<DeliveryFuture>) {
+    loop {
+        // Blocks until a frame is received
+        match channel_recv.recv().await {
+            Some(future) => match future.await {
+                Ok(_) => {
+                    trace!("Published event message");
+                    counter!(MESSAGES_PROCESSED).increment(1);
+                }
+                Err(e) => {
+                    error!("{:?}", e);
+                    counter!(
+                        FAILURES,
+                        &[failures::get_label(FailureKind::KafkaPublishFailed)]
+                    )
+                    .increment(1);
                 }
+            },
+            None => {
+                error!("Send-Frame Receiver Error");
```

Comment:
```suggestion
                info!("Send-Frame channel closed");
```

The channel closing is not an error, it will happen as part of the normal shutdown procedure.